### PR TITLE
Control default behaviour of error table through options

### PR DESCRIPTION
**Pull request by magedhennawy**
Solves [Display error table through options if mode is other than text](https://github.com/josdejong/jsoneditor/issues/1496)

### CODE REVIEW

**Comment by josdejong** [comment]
Thanks Maged for your PR. 

Right now the built-in behavior is: show the error table initially in modes `text` and `preview`, and not in `code` mode since that has icons in the gutter. 

It is a good idea to make this behavior customizable. Two thoughts:

1. This option should allow for the current behavior. I think we could make this option a callback like
    `showErrorTable: (mode: string) => boolean` to achieve that, or a list with modes like `showErrorTable: ['text', 'preview']` (maybe also allowing a boolean). 
3. I think the option should have a good default, I think the current behavior would be good.

What do you think?

**Comment by magedhennawy** [comment]
@josdejong Hey Jos, just updated it to match the showErrorTable array, and also possible a boolean. 
Please let me know what you think~

File: src/js/textmode.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ textmode.create = function (container, options = {}) {
   options.enableTransform = options.enableTransform !== false
   options.createQuery = options.createQuery || createQuery
   options.executeQuery = options.executeQuery || executeQuery
-
+  options.showErrorTable = options.showErrorTable || ['text']
```
Comment (josdejong) [inline src/js/textmode.js:53]: Can you change the default to `['text', 'preview']`?

**Review by josdejong — COMMENTED** [review]
That looks good👌. I made one small inline comment.

Can you also describe the new feature in `/docs/api.md`?

**Comment by magedhennawy** [comment]
All done @josdejong

**Comment by josdejong** [comment]
Thanks Maged! Merging your PR now. I see there is a minor linting issue, I'll fix that directly.

**Comment by josdejong** [comment]
Published now in `v9.10.0`, thanks again!

**Pull request merged**